Web Inspector: Crash when inspecting CSS Grid without defined columns or rows

patrickangle opened the pull request:
#### 05e01b57a4ecfed9f298898ed78f32bfd2456b94
<pre>
Web Inspector: Crash when inspecting CSS Grid without defined columns or rows
<a href="https://bugs.webkit.org/show_bug.cgi?id=256072">https://bugs.webkit.org/show_bug.cgi?id=256072</a>
rdar://108641874

Reviewed by Devin Rousso.

262869@main fixed issues with determining the authored grid track sizes, but in the process introduced a potential null
pointer deref due to us erroneously trying to get a reference to a RefPtr&apos;s value instead of getting its pointer for use
in a dynamic downcast.

* LayoutTests/inspector/dom/showFlexOverlay.html:
- Drive-by ensure we enable all options for flex overlays too so that those paths are exercises.

* LayoutTests/inspector/dom/showGridOverlay.html:
* Source/WebCore/inspector/InspectorOverlay.cpp:
(WebCore::authoredGridTrackSizes):

Canonical link: <a href="https://commits.webkit.org/263517@main">https://commits.webkit.org/263517@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2eb0a355f6993a7fbe459e6899d2584898c125a6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4814 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4936 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5104 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6329 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4942 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4806 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5107 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4908 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5183 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4895 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4984 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4313 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6342 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2472 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4308 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/9278 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4334 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4383 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5962 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4792 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3906 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4299 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/4304 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1188 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8362 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4661 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebCore/inspector/InspectorOverlay.cpp:
```diff
@@ -1441,7 +1441,7 @@ static Vector<String> authoredGridTrackSizes(Node* node, GridTrackSizingDirectio
     }
 
-    auto* cssValueList = dynamicDowncast<CSSValueList>(*cssValue);
+    auto* cssValueList = dynamicDowncast<CSSValueList>(cssValue.get());
```
do we need to `.get()`?  or can we just `dynamicDowncast<CSSValueList>(cssValue)`?

patrickangle replied:
Yeah, we need the `.get()` here, otherwise we try to downcast the RefPtr itself to CSSValueList, which doesn't compile 😅

dcrousso commented on LayoutTests/inspector/dom/showGridOverlay.html:
```diff
@@ -7,6 +7,14 @@
 {
     let suite = InspectorTest.createAsyncSuite("DOM.showGridOverlay");
 
+    const allEnabledOverlayOptions = {
```
i wonder if we should do this for the flex overlay too?

patrickangle replied:
Can't hurt.

patrickangle committed:
Web Inspector: Crash when inspecting CSS Grid without defined columns or rows
https://bugs.webkit.org/show_bug.cgi?id=256072
rdar://108641874

Reviewed by Devin Rousso.

262869@main fixed issues with determining the authored grid track sizes, but in the process introduced a potential null
pointer deref due to us erroneously trying to get a reference to a RefPtr's value instead of getting its pointer for use
in a dynamic downcast.

* LayoutTests/inspector/dom/showFlexOverlay.html:
- Drive-by ensure we enable all options for flex overlays too so that those paths are exercises.

* LayoutTests/inspector/dom/showGridOverlay.html:
* Source/WebCore/inspector/InspectorOverlay.cpp:
(WebCore::authoredGridTrackSizes):

Canonical link: https://commits.webkit.org/263517@main
